tests/utils/test_sampling_algorithms.py

```diff
@@ -31,3 +39,59 @@ def test_discrete_sampling(fraction, method):
         assert len(sampled) == len(
             sampled.drop_duplicates()
         ), "Undersized sampling did not return unique points."
+
+
+@given(
+    points=hnp.arrays(
+        dtype=float,
+        shape=hnp.array_shapes(min_dims=2, max_dims=2, min_side=1),
+        # Because of the square involved in the Euclidean distance computation,
+        # we limit the floating point range to avoid overflow problems
+        elements=st.floats(min_value=-1e100, max_value=1e100, allow_nan=False),
```

Comment:
"Limit the range" he says while using 100 digit numbers :D

Reply:
On a more serious note: Does this still work if the numerical precision is tuned down? I think we allow two types of numpy precision, right?

Reply:
Yeah, 1e100, the magic value :D But I have really good answer how to handle it differently. One **could** actually perform sanity checks in the form of guard clauses to prevent passing input that will blow up the numerics, but I have really never seen this in other code before...

Does it work with float32 -> we need to distinguish:
* The algorithm itself breaks above a certain numerical value. That happens for both float64 and float32, and we don't do anything against it.
* The **test** will always pass because it explicitly creates float64 as input, irrespective of how the environment variables are set. 

Does that answer the question?

Reply:
So *should* we do anything against the algorithm failing by introducing guard clauses?

Reply:
Unclear... So far, we don't have any check like this in the code, and catching these kinds of errors is really hard I think (for comparison: you also get all the cholesky and non-converged errors from botorch without them catching them). Not sure even how such a check would need to look like. You'd basically need to to check that you won't get an overflow/underflow **before** it happens or at least catch it after it happened. Overflow is probably easy (just check for np.inf), but underflow seems difficult

Reply:
Seems overflow is already correctly reported by `pairwise_distances`:
<img width="982" alt="image" src="https://github.com/user-attachments/assets/15912c2a-f4e8-4b08-9f3b-8c2febbfd3ab">
So let's close this thread